Packages/ClientRuntime/Sources/Waiters/WaiterErrors.swift

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Thrown when a `WaiterConfiguration` is constructed with invalid parameters.
+struct WaiterConfigurationError: Error {
```

Comment:
Should we make this public since the WaiterConfiguration initializer, that throws this error, is public? That way consumers could catch on the specific error if they wish

Reply:
Yes, this should definitely be public